src/test/scala/io/opentargets/etl/backend/HelpersTest.scala

```diff
@@ -26,13 +77,63 @@ class HelpersTest extends AnyFlatSpecLike with Matchers {
   it should "correctly rename columns in nested arrays" in {
     // given
     val structWithArray = testStruct
-      .add("d",ArrayType(new StructType()
-        .add("e",StringType)
-        .add("f",StringType)
-        .add("g",IntegerType)))
+      .add("d",
+           ArrayType(
+             new StructType()
+               .add("e", StringType)
+               .add("f", StringType)
+               .add("g", IntegerType)))
     // when
     val results = renameAllCols(structWithArray, renameFun)
     // then
-    assert(results(3).dataType.asInstanceOf[ArrayType].elementType.asInstanceOf[StructType].fieldNames.forall(_.head.isUpper))
+    assert(
+      results(3).dataType
+        .asInstanceOf[ArrayType]
+        .elementType
+        .asInstanceOf[StructType]
+        .fieldNames
+        .forall(_.head.isUpper))
   }
+
+  "applyFunToColumn" should "apply the function to the column and return a dataframe with the same column names as in the input dataframe" in {
+    import sparkSession.implicits._
+    // given
+    val df = Seq("UPPER").toDF("a")
+    // when
+    val results = df.transform(Helpers.applyFunToColumn("a", _, lower))
```

Comment:
please see above about why this is a bit cumbersome and not adding much to the way to process a column

Reply:
I see transform nice and useful when the implicated changes are really more like an algorithm or so.